### ISSUE TITLE
assignTaxonomy error in tax.out

### ISSUE THREAD

**Issue by topato** [benjjneb/dada2#64]
Hi,

First of all, thank you for the package. I'm quite excited to see an entire pipeline in R that can process community sequencing data in addition to the improvement in accuracy and sensitivity. 

I'm reporting a potential error-handling issue regarding `assignTaxonomy()`. Yesterday, I've finally had the chance to run a MiSeq dataset to try out DADA2. Everything went well until the taxonomy assignment stage. Here are the [sequences](https://dl.dropboxusercontent.com/u/374060/problem.txt) from the sequence table.

When I tried to classify the sequence table, this error occurred:
```
taxa_rdp <- assignTaxonomy(seqtab.nochim, paste0(path, "rdp_train_set_14.fa.gz"))
Error in tax.out[i, 1:length(taxes[[i]])] <- taxes[[i]] : 
  replacement has length zero
```
I have no problem when I subset parts of the table, including sequences shorter (240) than the average 272 bp. After digging through the sequence table I noticed a sequence distinctively unlike 16S rRNA gene was present:

`CCGCATTGTACATAAGGCTGGCACGGAACGCGATTTCAGGGATTTCATAAGCGGCACCAACGCGCCAGCCGACGCCATCCGCGCTCAGGTTCAGCCTGCCGGTGCCATCGAGCGCACCGTGTGTCAGCAACGGAACAAAGCCGTTGACCATGCGTGTCTTGAAGCCTTCGAGTTCCTGATAGGAAACGCCACCGATGAAACGCAGCGAGCCCTTGTCCGTAACATGGAACTTGTAGGAGCAGGTCAGGCCCCAGTCATCCGATGAAATCTTGGTTTCAATGTTCGAGTTGTCGCCGGCCCAATTGGAACCGGGCTTGGTGTGAACACCCCAAGGCTGGCTGTAGAAACC`

It appeared to be most similar to sequences from long chain fatty acid transporters in _Rhizobium_ spp. Removing the long sequence seemed to make the above error to go away with gg\_13\_8\_train\_set_97.fa.gz but not rdp\_train\_set\_14.fa.gz.

Am I missing something really obvious?

If it's due to "bad" sequences sneaking through, perhaps it might be a good idea for the taxonomy classifier to handle very occasional sequences that are not 16S rRNA gene by putting them into the "Others" category (if it isn't already the case).

Thanks,

Kevin Lee

**Comment by benjjneb** [comment]
Hi Kevin, thanks for this report. As you suspected, this error was triggered by non-16S sequences in the output that were unassigned at all taxonomic levels. It was a coding bug (i.e. not checking for that possibility) that caused this issue, and I just checked in a fix: 5b57f28a85cbb8ee4ce4fa78544e9d5f81b47452.

Now when every taxonomic level is assigned with `< minBoot` confidence, the output taxonomy table should correctly have `NA` at each taxonomic rank for that sequence (the equivalent of an "Other" category sequences) rather than error-ing out.

**Comment by topato** [comment]
Thanks! I can confirm it's working for me now.
